packages/models/src/tenant/protobufConverterFromV2.ts

```diff
@@ -0,0 +1,192 @@
+      throw new Error("Unspecified tenant mail kind");
+  }
+};
+
+export const fromTenantMailV2 = (input: TenantMailV2): TenantMail => ({
+  ...input,
+  id: input.id
+    ? input.id
+    : createHash("sha256").update(input.address).digest("hex"),
+  createdAt: new Date(Number(input.createdAt)),
+  kind: fromTenantMailKindV2(input.kind),
+});
+
+export const fromTenantFeatureV2 = (
+  input: TenantFeatureV2
+): TenantFeatureCertifier =>
+  match<TenantFeatureV2["sealedValue"], TenantFeatureCertifier>(
+    input.sealedValue
+  )
+    .with({ oneofKind: "certifier" }, ({ certifier }) => ({
+      type: "PersistentCertifier",
+      certifierId: certifier.certifierId,
+    }))
+    .with({ oneofKind: undefined }, () => {
+      throw new Error("Unable to deserialize TenantFeature");
+    })
+    .exhaustive();
+
+export const fromTenantVerifierV2 = (
+  input: TenantVerifierV2
+): TenantVerifier => ({
+  ...input,
+  verificationDate: new Date(Number(input.verificationDate)),
+  expirationDate: input.expirationDate
+    ? new Date(Number(input.expirationDate))
+    : undefined,
+  extensionDate: input.extensionDate
+    ? new Date(Number(input.extensionDate))
+    : undefined,
+});
+
+export const fromTenantRevokerV2 = (input: TenantRevokerV2): TenantRevoker => ({
+  ...input,
+  expirationDate: input.expirationDate
+    ? new Date(Number(input.expirationDate))
+    : undefined,
+  extensionDate: input.extensionDate
+    ? new Date(Number(input.extensionDate))
+    : undefined,
+  revocationDate: new Date(Number(input.revocationDate)),
+  verificationDate: new Date(Number(input.verificationDate)),
+});
+
+export const fromTenantAttributesV2 = (
+  input: TenantAttributeV2
+): TenantAttribute => {
+  const { sealedValue } = input;
+
+  switch (sealedValue.oneofKind) {
+    case "certifiedAttribute":
+      const { certifiedAttribute } = sealedValue;
+      return {
+        id: unsafeBrandId(certifiedAttribute.id),
+        assignmentTimestamp: new Date(
+          Number(certifiedAttribute.assignmentTimestamp)
+        ),
+        type: tenantAttributeType.CERTIFIED,
+      };
+    case "verifiedAttribute":
+      const { verifiedAttribute } = sealedValue;
+      return {
+        id: unsafeBrandId(verifiedAttribute.id),
+        assignmentTimestamp: new Date(
+          Number(verifiedAttribute.assignmentTimestamp)
+        ),
+        verifiedBy: verifiedAttribute.verifiedBy.map(fromTenantVerifierV2),
+        revokedBy: verifiedAttribute.revokedBy.map(fromTenantRevokerV2),
+        type: tenantAttributeType.VERIFIED,
+      };
+    case "declaredAttribute":
+      const { declaredAttribute } = sealedValue;
+      return {
+        id: unsafeBrandId(declaredAttribute.id),
+        assignmentTimestamp: new Date(
+          Number(declaredAttribute.assignmentTimestamp)
+        ),
+        type: tenantAttributeType.DECLARED,
+      };
+    case undefined:
+      throw genericError("Undefined attribute kind");
```

Comment:
In order to be safer and catch all invalid kind, this could be a default statement
```suggestion
    default:
      throw genericError(`Invalid attribute kind: ${sealedValue.oneofKind}`);
```

Reply:
Updated